### PR TITLE
Branch.io updates

### PR DESCRIPTION
**Pull request by ameypv-rudder**
## What do these changes do?

> Updated Branch documentation
> Added device mode integration for iOS v2.

## Type of documentation

- [ ] New documentation
- [x] Documentation update
- [ ] Hotfix

### CODE REVIEW

File: docs/destinations/streaming-destinations/branchio.mdx [hunk 1]
```diff
@@ -24,32 +24,40 @@ Before getting started, please determine whether the platform you are sending yo
 To know more about the difference between cloud mode and device mode in RudderStack, refer to the <Link to="/destinations/rudderstack-connection-modes/">RudderStack Connection Modes</Link> guide.
 </div>
 
-In order to start sending data to Branch, you will first need to add it as a destination to the source from which you are sending the event data.
+<div class="warningBlock">
```
Comment (ishachopra0401) [inline docs/destinations/streaming-destinations/branchio.mdx:27]: Not quite sure about its relevance, as its put across from the table above

File: docs/destinations/streaming-destinations/branchio.mdx [hunk 2]
```diff
@@ -283,10 +316,8 @@ The following table lists the mapping of the accepted property keys common to al
...
 | `condition`              | `$condition`            |
 
-## FAQs
-
-### Where can I get the Branch Key from?
-
-You can retrieve the Branch key from the settings page of the [Branch dashboard](https://dashboard.branch.io/#/settings). For more information, please check the [Getting Started](#getting-started) section above.
+## FAQ
 
+### Where can I find the Branch key?
```
Comment (ishachopra0401) [inline docs/destinations/streaming-destinations/branchio.mdx:321]: Its as good as present as being removed :p 
Lets remove it or add some detailed steps with a screenshot. Thoughts?

File: docs/destinations/streaming-destinations/branchio.mdx [hunk 3]
```diff
@@ -24,32 +24,40 @@ Before getting started, please determine whether the platform you are sending yo
...
 
-<div class="warningBlock">
+## Connection settings
+
+To successfully configure Branch as a destination, you will need to add your Branch key by going to the **Settings** page of your Branch [dashboard](https://dashboard.branch.io/#/settings).
 
-Please note that RudderStack does not support sending data to Branch on both device or cloud mode for the web.
+<img src="../../assets/event-stream-destinations/branchio-connection-settings.png" alt="Branch connection settings"/>
 
+- **Client-side Events Filtering**: This setting lets you specify which events should be blocked or allowed to flow through to ConvertFlow.
```
Comment (ishachopra0401) [inline docs/destinations/streaming-destinations/branchio.mdx:42]:
```suggestion
- **Client-side Events Filtering**: This setting lets you specify which events should be blocked or allowed to flow through to Branch.
```

File: docs/destinations/streaming-destinations/branchio.mdx [hunk 4]
```diff
@@ -24,32 +24,41 @@ Before getting started, please determine whether the platform you are sending yo
...
+Once you have confirmed that the source platform supports sending events to Branch, follow these steps:
 
-<img src="../../assets/screenshot-2020-02-23-at-3.25.23-pm.png" /><span class="imageTitle">BranchIO Dashboard Settings</span>
+1. From your [RudderStack dashboard](https://app.rudderstack.com/), add a source. Then, from the list of destinations, select **Branch Metrics**.
+2. Assign a name to the destination and click **Continue**.
 
-<div class="warningBlock">
+## Connection settings
+
+To successfully configure Mixpanel as a destination, you will need to configure the following settings:
```
Comment (ishachopra0401) [inline docs/destinations/streaming-destinations/branchio.mdx:38]:
```suggestion
To successfully configure Branch as a destination, you will need to configure the following settings: 
```

File: docs/destinations/streaming-destinations/branchio.mdx (hunk 2 above)
Comment (ameypv-rudder) [reply docs/destinations/streaming-destinations/branchio.mdx:318]: Makes sense! Updated the FAQ to add more details.

File: docs/destinations/streaming-destinations/branchio.mdx (hunk 1 above)
Comment (ameypv-rudder) [reply docs/destinations/streaming-destinations/branchio.mdx]: Let's remove it entirely. Made the change.

**Review by ishachopra0401 — APPROVED** [review]

**Pull request merged**